SOLR-14702: Remove oppressive language (part1)

MarcusSorealheis opened the pull request:
<!--
_(If you are a project committer then you may remove some/all of the following template.)_

Before creating a pull request, please file an issue in the ASF Jira system for Lucene or Solr:

* https://issues.apache.org/jira/projects/LUCENE
* https://issues.apache.org/jira/projects/SOLR

You will need to create an account in Jira in order to create an issue.

The title of the PR should reference the Jira issue number in the form:

* LUCENE-####: <short description of problem or changes>
* SOLR-####: <short description of problem or changes>

LUCENE and SOLR must be fully capitalized. A short description helps people scanning pull requests for items they can work on.

Properly referencing the issue in the title ensures that Jira is correctly updated with code review comments and commits. -->


# Description

Please refer to this PR: https://github.com/apache/lucene-solr/pull/1711

# Solution

Replaces master with primary and slave with secondary.

# Tests

Run standard tests.


# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [x] I have run `ant precommit` and the appropriate test suite.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

MarcusSorealheis commented on solr/core/src/java/org/apache/solr/util/hll/HLL.java:
```diff
@@ -47,7 +47,7 @@
  *   <li><a href="https://github.com/aggregateknowledge/js-hll">js-hll</a></li>
  * </ul>
- * when <a href="https://github.com/aggregateknowledge/postgresql-hll/blob/master/STORAGE.markdown">properly serialized</a>.
+ * when <a href="https://github.com/aggregateknowledge/postgresql-hll/blob/secondaryy/STORAGE.markdown">properly serialized</a>.
```
is a typo. good catch sorry.

MarcusSorealheis commented:
taking this out of Draft status. 👨🏾‍💻

MarcusSorealheis commented:
https://media.giphy.com/media/Y4swGdfY94hHMUrwcZ/giphy.gif

murblanc commented on solr/contrib/dataimporthandler/src/test/org/apache/solr/handler/dataimport/TestContentStreamDataSource.java:
```diff
@@ -120,7 +120,7 @@ public void testCommitWithin() throws Exception {
     
     /**
-     * if masterPort is null, this instance is a master -- otherwise this instance is a slave, and assumes the master is
+     * if masterPort is null, this instance is a master -- otherwise this instance is a secondary, and assumes the master is
```
missed "master" references in comment (3)

murblanc commented:
PR description says:
> Replaces master with leader and slave with follower.

Actual replacement is primary/secondary.

I don't see the renamed `.png` files. Are they part of a different PR?

murblanc commented on solr/core/src/java/org/apache/solr/security/JWTVerificationkeyResolver.java:
```diff
@@ -106,7 +106,7 @@ public Key resolveKey(JsonWebSignature jws, List<JsonWebStructure> nestingContex
       }
 
-      // Add all keys into a master list
+      // Add all keys into a primary list
```
This occurrence of "master" is unrelated to master/slave replication. Maybe simply remove the word "master" or replace it with "reference" but "primary" doesn't really make sense.

MarcusSorealheis replied:
yea, that was a screw up of mine. I would say this can stay master as I have no inherent problem with the word `master` outside of the context of `master/slave`.

However, I agree that probably best to remove this one in this case because others do.  I'll change it to reference.

murblanc commented on solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java:
```diff
@@ -1241,33 +1241,33 @@ public void inform(SolrCore core) {
     @SuppressWarnings({"rawtypes"})
-    NamedList master = (NamedList) initArgs.get("master");
-    boolean enableMaster = isEnabled( master );
+    NamedList primary = (NamedList) initArgs.get("primary");
```
Isn't this a communication between different Solr nodes?
If the current node got updated to new vocabulary primary/secondary but the remote node did not and did send data using master/slave, we might run into issues.

It might be desirable (if the above is true) to support both primary and master as return values (and secondary and slave) until a subsequent version of Solr not expected to work with nodes running code using master/slave.

murblanc replied:
Also same issue in the other direction: if an updated node sends "primary" to a non updated node expecting "master", the non updated node will fail. Maybe we need to send the same data under both "primary" and "master" and the same data under both "secondary" and "slave", and accept either one on the receiving end.

I'm not familiar with how this code exactly works so might be off here, but this rename is not internal so must use caution (and test with renamed nodes talking to non renamed nodes, on both roles).

MarcusSorealheis replied:
I'm going to look through it. Hopefully I can grok it. Not sure, though. If I can find a more backcompat approach, I will drop it in the PR. Open to suggestions from others.

MarcusSorealheis replied:
`initArgs.containsKey("primary") ? initArgs.get("primary") : initArgs.get("master")`?

murblanc replied:
As @chatman commented in the general discussion, maybe these changes happening in a new major version branch does no require them to be backward/forward compatible?

If 8x and 9.0 servers are expected to talk and understand each other, the suggestion looks good on the receiving end. We need something on the sending end as well (send both values, because "primary" being sent to a 8x server expecting only "master" will not be understood).

MarcusSorealheis replied:
I changed it to this: `NamedList primary = (NamedList)(initArgs.get("primary") != null ? initArgs.get("primary") : initArgs.get("master"));`

janhoy replied:
We still promote rolling upgrade as the way to do upgrades in https://lucene.apache.org/solr/guide/8_6/upgrading-a-solr-cluster.html with no disclaimer about major versions. In the past we have strived to make a new major version be able to talk to previous major version. Could also let 8.7 support both and have the back-compat and require users to upgrade to 8.7 or later 8.x before doing rolling upgrade to 9.0. Or document that 8.x to 9.0 will not support rolling upgrade...

chatman replied:
Rolling upgrade across major versions isn't supported, and I don't see any reason to bend over backwards to support that. We can just document the changes and move on. This PR should be merged as soon as possible because it is a very large PR and can fall out of date very quickly.

janhoy replied:
> Rolling upgrade across major versions isn't supported
I hope that is clearly documented somewhere? The only refguide page I could find on upgrading is the rolling one :)

Btw - since when was the size of a PR a good reason for rushing it into the codebase?

chatman replied:
>  I hope that is clearly documented somewhere? The only refguide page I could find on upgrading is the rolling one :)

If not documented, it should be. It has never been supported explicitly.

> Btw - since when was the size of a PR a good reason for rushing it into the codebase?

Hahah ;-) Ever since you displayed intent to sabotage/stall this effort (by requesting Marcus to go back to the solr-user list and discussing it there) :-) Anyway, jokes aside, this PR is quite close to being ready, except for the backcompat issue (which I don't think matters at all).

murblanc replied:
> except for the backcompat issue (which I don't think matters at all).

What's the real issue? Let's accept as input "master" or "primary" in one case and "slave" or "secondary" in the other, as well as send both values over the wire as output.
This will make new code and old code compatible (old talking to new and new talking to old). Once only new code is running, we can safely remove sending or accepting the offending terms.

tflobbe replied:
I don't think this discussion is resolved?
> What's the real issue? Let's accept as input "master" or "primary" in one case and "slave" or "secondary" in the other, as well as send both values over the wire as output.
This will make new code and old code compatible (old talking to new and new talking to old). Once only new code is running, we can safely remove sending or accepting the offending terms.

+1, this should allow us to be backward compatible (and I'd even suggest we keep that for the life of 9 to make upgrades easy)

MarcusSorealheis replied:
That's what I added I thought @tflobbe Can you let me know where there other places where I should anticipate receiving `slave` or `master`

ctargett replied:
> Rolling upgrade across major versions isn't supported, and I don't see any reason to bend over backwards to support that. 

Not to derail this effort at all, which I think is very important, but we *do* claim to support rolling upgrades across major versions, see for example https://lucene.apache.org/solr/guide/8_6/major-changes-in-solr-8.html#rolling-upgrades-with-solr-8. The page on upgrading a Solr cluster is really only meant for SolrCloud users (that's what was meant by "cluster" when it was written, which is actually incorrect since you can have a "cluster" without using SolrCloud), but it doesn't say not to do a rolling upgrade across major versions because we have other docs that tell you how to do exactly that.

I don't think it's a major issue though, we just need to clearly spell out the proper upgrade steps for 9.0 (which doesn't have to be part of this PR). There is going to be a lot for people to do (if they're using removed features and need to install plugins), this isn't going to add much IMO.

MarcusSorealheis replied:
> > except for the backcompat issue (which I don't think matters at all).
> 
> What's the real issue? Let's accept as input "master" or "primary" in one case and "slave" or "secondary" in the other, as well as send both values over the wire as output.
> This will make new code and old code compatible (old talking to new and new talking to old). Once only new code is running, we can safely remove sending or accepting the offending terms.

The real issue here is that the terms should've been removed years ago, or never used. They are super stressful to interact with and every time I combed through my changes here it was really frustrating. I was in college when the project was released. These terms were not used at Grinnell College. I could give a history lesson if you'd like to understand, but I think we should align with the standards put forth by the IETF and expressed in the Apache Code of Conduct at a minimum. We should also remember to prioritize community over code. We shouldn't even be looking to preserve rolling upgrades in favor of keeping such stressful nomenclature. There's so much changing in 9 already that is backward incompatible.

MarcusSorealheis commented:
> PR description says:
> 
> > Replaces master with leader and slave with follower.
> 
> Actual replacement is primary/secondary.
> 
I've fixed the description. That was a vestige of my initial PR where I ran into problems when I tried to run/understand the changes. 

> I don't see the renamed `.png` files. Are they part of a different PR?

On the PNG's, I've fixed them. I was aware of them and forgot to address. I think subsequent PRs will follow, but haven't determined the scope yet. I want to move away from surface level fixes that have deeply emotional implications for code readers and developers on a social level to surface level fixes that have deeply emotional implications for code readers on a technical level. Lots of words there. **Better developer experience.**

chatman commented on solr/solr-ref-guide/src/query-settings-in-solrconfig.adoc:
```diff
@@ -203,7 +203,7 @@ This setting controls whether search requests for which there is not a currently
 === maxWarmingSearchers
 
-This parameter sets the maximum number of searchers that may be warming up in the background at any given time. Exceeding this limit will raise an error. For read-only slaves, a value of two is reasonable. Masters should probably be set a little higher.
+This parameter sets the maximum number of searchers that may be warming up in the background at any given time. Exceeding this limit will raise an error. For read-only secondaries, a value of two is reasonable. Primarys should probably be set a little higher.
```
Instead of "secondaries" and "primaries" (or "primarys"), lets call them "primary servers" and "secondary servers" in the reference guide.

MarcusSorealheis replied:
ok that's a good suggestion. Primarys was another typo. I'll add it.

MarcusSorealheis replied:
I've made this change @chatman

MarcusSorealheis commented:
After all the recent changes, making sure the tests pass

MarcusSorealheis commented:
@chatman it looks like the tests pass when I run the TestReplicationHandler on its own but it fails when I run ant test locally: 
![image](https://user-images.githubusercontent.com/2353608/89136063-91afb700-d4e6-11ea-9f25-1147f8825815.png)

MarcusSorealheis commented:
This also passes: 
![image](https://user-images.githubusercontent.com/2353608/89136223-6bd6e200-d4e7-11ea-8c1e-96a87de73f64.png)

MarcusSorealheis commented:
I've gone through and tested these classes individually and they all pass, yet these tests fail when run with the entire project:

```[junit4] Tests with failures [seed: 2F07E49425A06B54]:
   [junit4]   - org.apache.solr.cloud.api.collections.ShardSplitTest.testSplitAfterFailedSplit
   [junit4]   - org.apache.solr.handler.TestReplicationHandler.testRateLimitedReplication
   [junit4]   - org.apache.solr.handler.TestReplicationHandler.doTestRepeater
   [junit4]   - org.apache.solr.handler.TestReplicationHandler.doTestStressReplication
   [junit4]   - org.apache.solr.handler.TestReplicationHandler.doTestIndexFetchWithPrimaryUrl
   [junit4]   - org.apache.solr.cloud.api.collections.ShardSplitTest (suite)```

This suggests to me that there must be some shared state that exists between other tests in the suite or there is flakiness. I don't know anything about the flakiness. Inly know that there are flaky tests. Does anyone have any idea what I'm missing. CC @chatman @murblanc

MarcusSorealheis commented:
This suggests to me that there must be some shared state that exists between other tests in the suite or there is flakiness. I don't know anything about the flakiness. Inly know that there are flaky tests. Does anyone have any idea what I'm missing. CC @chatman @murblanc

MarcusSorealheis commented:
I ran it again and shard split tests didn't fail so there seems to be some inconsistency there. ReplicationHandler tests still failed. I'm digging as to what they test to see if I can determine why they don't fail when isolated but do fail when run together.

See below:

```
[junit4]   - org.apache.solr.handler.TestReplicationHandler.doTestStressReplication
[junit4]   - org.apache.solr.handler.TestReplicationHandler.doTestRepeater
[junit4]   - org.apache.solr.handler.TestReplicationHandler.testRateLimitedReplication
[junit4]   - org.apache.solr.handler.TestReplicationHandler.doTestIndexFetchWithPrimaryUrl
```

So strange

MarcusSorealheis commented:
Looks like `doTestStressReplicaion` is a confirmed Bad Apple as defined here: https://issues.apache.org/jira/browse/SOLR-12028

In other words, moving on to check the others. I'll batch my comments accordingly.

MarcusSorealheis commented:
I cannot get the additional three tests to fail with consistency when isolated, but I cannot say with any certainty that they are not an issue. A second (maybe third) set of eyes would be really helpful here. It's really tricky for me to understand why they are failing the test config files look good afaik. let me know if this warrants further investigation.

murblanc commented:
> I cannot get the additional three tests to fail with consistency when isolated, but I cannot say with any certainty that they are not an issue. A second (maybe third) set of eyes would be really helpful here. It's really tricky for me to understand why they are failing the test config files look good afaik. let me know if this warrants further investigation.

Did you run these tests with same scrutiny before your changes? Maybe behavior is similar?

noblepaul commented:
>This PR should be merged as soon as possible because it is a very large PR and can fall out of date very quickly.

@chatman . We should get things correct before merging it. This has to be vetted properly

MarcusSorealheis commented:
> Did you run these tests with same scrutiny before your changes? Maybe behavior is similar?

I had not. I am checking now.

ctargett commented:
I looked at the Ref Guide changes, +1 on those from me (but I can't make it a blanket +1 since I can't comment on all the other code changes).

MarcusSorealheis commented:
in my most recent test, the previously failing tests all passed. Thanks @ErickErickson.

However, a new set of tests failed.  Tests with failures [seed: 358D229C22BF813]:
  ```
 [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.test
   [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testWhitebox
   [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testCreepThenBite {seed=[358D229C22BF813:D4908E36E9A995B2]}
   [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testCreepThenBite {seed=[358D229C22BF813:60C632CADD6B5E9E]}
   [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testCreepThenBite {seed=[358D229C22BF813:EE2FA4168CAF1655]}
   [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testCreepThenBite {seed=[358D229C22BF813:DFC10FF3E0514D7C]}
   [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testCreepThenBite {seed=[358D229C22BF813:93008A5E41598DC7]}
   [junit4]   - org.apache.solr.schema.TestBulkSchemaConcurrent (suite)
```

MarcusSorealheis commented:
Then they passed with the same seed.

BUILD SUCCESSFUL
```
Total time: 2 minutes 43 seconds
Marcuss-MacBook-Pro:lucene-solr marcussorealheis$ history 2
  600   ant test  -Dtestcase=CloudExitableDirectoryReaderTest -Dtests.method=testCreepThenBite -Dtests.seed=358D229C22BF813 -Dtests.slow=true -Dtests.badapples=true -Dtests.locale=sr-Cyrl -Dtests.timezone=America/Argentina/San_Juan -Dtests.asserts=true -Dtests.file.encoding=UTF-8
```

MarcusSorealheis commented:
all tests pass: 

![image](https://user-images.githubusercontent.com/2353608/89250404-8c776880-d5c9-11ea-9dc4-b3dfebfed92d.png)

Can we agree that I can add the back-compat changes to support rolling upgrades in a subsequent PR @tflobbe @ctargett @thelabdude @murblanc @noblepaul @elyograg @janhoy @chatman

tflobbe commented on solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java:
```diff
@@ -1768,13 +1770,13 @@ private static Long readIntervalNs(String interval) {
+  // in case of TLOG replica, if primaryVersion = zero, don't do commit
   // otherwise updates from current tlog won't copied over properly to the new tlog, leading to data loss
-  public static final String SKIP_COMMIT_ON_MASTER_VERSION_ZERO = "skipCommitOnMasterVersionZero";
+  public static final String SKIP_COMMIT_ON_MASTER_VERSION_ZERO = "skipCommitOnPrimaryVersionZero";
```
Missed updating the constant name? The value on the other side is used as a parameter between instances, we should be able to send the previous value given a back compat mode (so that old instances can read what new instances are sending)

tflobbe commented on solr/core/src/java/org/apache/solr/handler/component/FacetComponent.java:
```diff
@@ -1033,9 +1033,9 @@ private void refinePivotFacets(ResponseBuilder rb, ShardRequest sreq) {
-        PivotFacet masterPivotFacet = fi.pivotFacets.get(pivotFacetResponseFromShard.getKey());
-        masterPivotFacet.mergeResponseFromShard(shardNumber, rb, pivotFacetResponseFromShard.getValue());  
-        masterPivotFacet.removeAllRefinementsForShard(shardNumber);
+        PivotFacet primaryPivotFacet = fi.pivotFacets.get(pivotFacetResponseFromShard.getKey());
+        primaryPivotFacet.mergeResponseFromShard(shardNumber, rb, pivotFacetResponseFromShard.getValue());
+        primaryPivotFacet.removeAllRefinementsForShard(shardNumber);
```
This has nothing to do with master/slave replication. Is this something we want to change too? If so, I guess this is an aggregation?

tflobbe commented on solr/core/src/java/org/apache/solr/handler/component/StatsValuesFactory.java:
```diff
@@ -182,7 +182,7 @@ protected AbstractStatsValues(StatsField statsField) {
       //
       // part of the complexity here being that the StatsValues API serves two
-      // masters: collecting concrete Values from things like DocValuesStats and
+      // primaries: collecting concrete Values from things like DocValuesStats and
```
Interestingly, this has nothing to do with replication, no clue what this means here

tflobbe commented on solr/core/src/java/org/apache/solr/search/stats/ExactSharedStatsCache.java:
```diff
@@ -40,7 +40,7 @@
   private final Map<String,Map<String,TermStats>> perShardTermStats = new ConcurrentHashMap<>();
   private final Map<String,Map<String,CollectionStats>> perShardColStats = new ConcurrentHashMap<>();
-  // global stats synchronized from the master
+  // global stats synchronized from the primary
```
same as above

tflobbe commented on solr/core/src/java/org/apache/solr/search/stats/LRUStatsCache.java:
```diff
@@ -65,7 +65,7 @@
   private final Map<String,Map<String,CollectionStats>> perShardColStats = new ConcurrentHashMap<>();
   
-  // global stats synchronized from the master
+  // global stats synchronized from the primary
```
same as above

tflobbe commented on solr/core/src/java/org/apache/solr/search/stats/StatsCache.java:
```diff
@@ -176,7 +176,7 @@ public void mergeToGlobalStats(SolrQueryRequest req,
 
   /**
-   * Receive global stats data from the master and update a local cache of global stats
+   * Receive global stats data from the primary and update a local cache of global stats
```
same as above

tflobbe commented on solr/core/src/test/org/apache/solr/cloud/cdcr/CdcrReplicationHandlerTest.java:
```diff
@@ -59,14 +59,14 @@ public void distribSetUp() throws Exception {
 
   /**
-   * Test the scenario where the slave is killed from the start. The replication
+   * Test the scenario where the secondary is killed from the start. The replication
```
Everything in this class is SolrCloud-related, not legacy replication

tflobbe commented on solr/solr-ref-guide/src/cloud-screens.adoc:
```diff
@@ -23,7 +23,7 @@ This screen provides status information about each collection & node in your clu
 [NOTE]
 ====
-The "Cloud" menu option is only available on Solr instances running in <<getting-started-with-solrcloud.adoc#getting-started-with-solrcloud,SolrCloud mode>>. Single node or master/slave replication instances of Solr will not display this option.
+The "Cloud" menu option is only available on Solr instances running in <<getting-started-with-solrcloud.adoc#getting-started-with-solrcloud,SolrCloud mode>>. Single node or primary/secondary replication instances of Solr will not display this option.
```
Should we say "legacy replication" instead? (especially if we decide to go with leader/follower)

MarcusSorealheis replied:
You are the only one truly **adamant** about changing to leader follower that I have seen thus far, so I'm not sure we will go that route. Unless, of course, you are willing to make the changes in that branch. You have a had start. :)

tflobbe replied:
[I'm not?](http://mail-archives.apache.org/mod_mbox/lucene-solr-user/202006.mbox/%3cCAKiERN77Q+LDCKMjhKcvitSLeY6PTikVnSErHv-Xa1RYsiqnrA@mail.gmail.com%3e). We can do a sweeping change after we are done with this PR and before merging. Shouldn't be too bad. Also, my comment is about the "mode", which has some other arguments also in that thread.

MarcusSorealheis replied:
Ok

MarcusSorealheis replied:
I'd never seen that email

MarcusSorealheis replied:
I don't think I'm on that list.

tflobbe commented on solr/solr-ref-guide/src/collection-specific-tools.adoc:
```diff
@@ -24,7 +24,7 @@ In the left-hand navigation bar, you will see a pull-down menu titled "Collectio
 The "Collection Selector" pull-down menu is only available on Solr instances running in <<solrcloud.adoc#solrcloud,SolrCloud mode>>.
 
-Single node or master/slave replication instances of Solr will not display this menu, instead the Collection specific UI pages described in this section will be available in the <<core-specific-tools.adoc#core-specific-tools,Core Selector pull-down menu>>.
+Single node or primary/secondary replication instances of Solr will not display this menu, instead the Collection specific UI pages described in this section will be available in the <<core-specific-tools.adoc#core-specific-tools,Core Selector pull-down menu>>.
```
same as above

tflobbe commented on solr/solr-ref-guide/src/coreadmin-api.adoc:
```diff
@@ -19,7 +19,7 @@
 The Core Admin API is primarily used under the covers by the <<collections-api.adoc#collections-api,Collections API>> when running a <<solrcloud.adoc#solrcloud,SolrCloud>> cluster.
 
-SolrCloud users should not typically use the CoreAdmin API directly, but the API may be useful for users of single-node or master/slave Solr installations for core maintenance operations.
+SolrCloud users should not typically use the CoreAdmin API directly, but the API may be useful for users of single-node or primary/secondary Solr installations for core maintenance operations.
```
same as above

tflobbe commented on solr/solr-ref-guide/src/distributed-search-with-index-sharding.adoc:
```diff
@@ -61,7 +61,7 @@ The following components support distributed search:
 === Shards Whitelist
 
-The nodes allowed in the `shards` parameter is configurable through the `shardsWhitelist` property in `solr.xml`. This whitelist is automatically configured for SolrCloud but needs explicit configuration for master/slave mode. Read more details in the section <<distributed-requests.adoc#configuring-the-shardhandlerfactory,Configuring the ShardHandlerFactory>>.
+The nodes allowed in the `shards` parameter is configurable through the `shardsWhitelist` property in `solr.xml`. This whitelist is automatically configured for SolrCloud but needs explicit configuration for primary/secondary mode. Read more details in the section <<distributed-requests.adoc#configuring-the-shardhandlerfactory,Configuring the ShardHandlerFactory>>.
```
same as above

tflobbe commented on solr/solr-ref-guide/src/replication-screen.adoc:
```diff
@@ -16,10 +16,10 @@
-The Replication screen shows you the current replication state for the core you have specified. <<solrcloud.adoc#solrcloud,SolrCloud>> has supplanted much of this functionality, but if you are still using Master-Slave index replication, you can use this screen to:
+The Replication screen shows you the current replication state for the core you have specified. <<solrcloud.adoc#solrcloud,SolrCloud>> has supplanted much of this functionality, but if you are still using Primary-Secondary index replication, you can use this screen to:
 
 . View the replicatable index state. (on a master node)
```
missed?

MarcusSorealheis replied:
Missed what here?

tflobbe replied:
"View the replicatable index state. (on a master node)"

MarcusSorealheis commented on solr/core/src/java/org/apache/solr/cloud/ReplicateFromLeader.java:
```diff
@@ -76,12 +76,12 @@ public void startReplication(boolean switchTransactionLog) throws InterruptedExc
-      slaveConfig.add(ReplicationHandler.SKIP_COMMIT_ON_MASTER_VERSION_ZERO, switchTransactionLog);
-      slaveConfig.add("pollInterval", pollIntervalStr);
+      NamedList<Object> secondaryConfig = new NamedList<>();
+      secondaryConfig.add("fetchFromLeader", Boolean.TRUE);
+      secondaryConfig.add(ReplicationHandler.SKIP_COMMIT_ON_MASTER_VERSION_ZERO, switchTransactionLog);
+      secondaryConfig.add("pollInterval", pollIntervalStr);
       NamedList<Object> replicationConfig = new NamedList<>();
-      replicationConfig.add("slave", slaveConfig);
+      replicationConfig.add("secondary", secondaryConfig);
```
thanks

MarcusSorealheis commented on solr/core/src/java/org/apache/solr/cloud/RecoveryStrategy.java:
```diff
@@ -239,7 +239,7 @@ final private void replicate(String nodeName, SolrCore core, ZkNodeProps leaderp
     ModifiableSolrParams solrParams = new ModifiableSolrParams();
-    solrParams.set(ReplicationHandler.MASTER_URL, leaderUrl);
+    solrParams.set(ReplicationHandler.PRIMARY_URL, leaderUrl);
     solrParams.set(ReplicationHandler.SKIP_COMMIT_ON_MASTER_VERSION_ZERO, replicaType == Replica.Type.TLOG);
```
oof. good catch

MarcusSorealheis replied:
On second thought, this is a reference to the replication handler. One of the view links between the old way and the new one.

MarcusSorealheis commented on solr/core/src/java/org/apache/solr/handler/CdcrRequestHandler.java:
```diff
@@ -733,12 +733,12 @@ private void handleBootstrapStatus(SolrQueryRequest req, SolrQueryResponse rsp)
 
   static class BootstrapCallable implements Callable<Boolean>, Closeable {
-    private final String masterUrl;
+    private final String primaryUrl;
```
this whole file should go away, but that's a PR for another night...but in progress! 😍

MarcusSorealheis commented on solr/core/src/java/org/apache/solr/handler/IndexFetcher.java:
```diff
@@ -236,27 +236,27 @@ public IndexFetcher(@SuppressWarnings({"rawtypes"})final NamedList initArgs, fin
-    Object skipCommitOnMasterVersionZero = initArgs.get(SKIP_COMMIT_ON_MASTER_VERSION_ZERO);
-    if (skipCommitOnMasterVersionZero != null && skipCommitOnMasterVersionZero instanceof Boolean) {
-      this.skipCommitOnMasterVersionZero = (boolean) skipCommitOnMasterVersionZero;
+    Object skipCommitOnPrimaryVersionZero = initArgs.get(SKIP_COMMIT_ON_MASTER_VERSION_ZERO);
```
I think this is the only one that needs to change, possibly. Still stitching it all together.

MarcusSorealheis replied:
It doesn't have to handle back-compat in this PR. I can commit to it prior to 9.0's release and even prioritize the work before opening PRs on the other work I am doing.

MarcusSorealheis commented on solr/core/src/java/org/apache/solr/handler/IndexFetcher.java:
```diff
@@ -404,35 +404,35 @@ IndexFetchResult fetchLatestIndex(boolean forceReplication, boolean forceCoreRel
 
-      log.info("Master's generation: {}", latestGeneration);
-      log.info("Master's version: {}", latestVersion);
+      log.info("Primary's generation: {}", latestGeneration);
+      log.info("Primary's version: {}", latestVersion);
```
gothca

MarcusSorealheis replied:
Shawn suggested that we shouldn't use the same terminology for the legacy architecture. I've updated the first PR in any event. At this time, keeping the same terminology seems to be the right way to go unless I rewrite the replication handler. There's an argument for that because I'm sure there are areas where we could improve, but it won't happen in this PR.

MarcusSorealheis commented:
I suppose I should convert this back to draft. I don't think it all the comments makes sense, but there is one place I think it does make sense to preserve the name: `ReplicationHandler`.  I didn't see the mail that went out from Anshum originally where there was some discussion. I started on this effort because I was sick of seeing things in the code base. I need to do some more investigation. I don't want to rush this out given the complexity and all the moving parts. I need to figure out if it makes sense to refactor some of these classes to separate concerns. Probably not because I don't want to add any duplicated logic unless there's a better reason than names. I'll come back sometime tomorrow, probably at night.

tflobbe commented:
One way to get back compatibility and rolling upgrades could be to make 9.x code be able to read previous formats, but write new format, and make 8.x (since 8.7) read new and old, but write old? Anyone wanting to do a rolling upgrade to 9 would have to be on at least 8.7. Rolling upgrades to 8.7 would still work.
All the code other than the requests/responses could be changed in 8_x branch, in addition to master.

noblepaul commented:
+1 to @tflobbe 

We should make the change not break rolling restarts. I like the suggestion

MarcusSorealheis commented:
PLEASE DO NOT MERGE — I've reverted back to the initial approach of using Leader/Follower because of various dependencies.

MarcusSorealheis commented:
Closing this PR in favor of the other: https://github.com/apache/lucene-solr/pull/1711

MarcusSorealheis commented:
@tflobbe I like the suggestion. Can we save it for the subsequent PR?

ErickErickson commented:
About non-reproducible test failures. See: http://fucit.org/solr-jenkins-reports/failure-report.html

Hoss rolls up the last 7 day’s failures from various Jenkins jobs. If a non-reproducible failure
shows up on that list as this one does it’s often an indication of a flaky test. 

This is actually an embarrassing  situation to be in, I have some hope that Mark Miller’s work 
will get incorporated sometime and we’ll have many fewer of these since they’re a PITA.

Erick

> On Aug 4, 2020, at 4:35 AM, Marcus <notifications@github.com> wrote:
> 
> 
> @tflobbe I like the suggestion. Can we save it for the subsequent PR?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
>

ErickErickson commented:
Marcus:

Also, that _particular_ test started failing a lot more recently, see: https://issues.apache.org/jira/browse/SOLR-14684

So I wouldn’t worry about it...

> On Aug 3, 2020, at 9:10 PM, Marcus <notifications@github.com> wrote:
> 
> 
> in my most recent test, the previously failing tests all passed. Thanks @ErickErickson.
> 
> However, a new set of tests failed. Tests with failures [seed: 358D229C22BF813]:
> 
> [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.test
>  [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testWhitebox
>  [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testCreepThenBite {seed=[358D229C22BF813:D4908E36E9A995B2]}
>  [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testCreepThenBite {seed=[358D229C22BF813:60C632CADD6B5E9E]}
>  [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testCreepThenBite {seed=[358D229C22BF813:EE2FA4168CAF1655]}
>  [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testCreepThenBite {seed=[358D229C22BF813:DFC10FF3E0514D7C]}
>  [junit4]   - org.apache.solr.cloud.CloudExitableDirectoryReaderTest.testCreepThenBite {seed=[358D229C22BF813:93008A5E41598DC7]}
>  [junit4]   - org.apache.solr.schema.TestBulkSchemaConcurrent (suite)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
>

MarcusSorealheis commented:
closing this in favor of the finished product: https://github.com/apache/lucene-solr/pull/1711
